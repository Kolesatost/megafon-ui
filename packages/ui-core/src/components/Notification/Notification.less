@import '~styles/base';

@b: e('.mfui-notification');

@{b} {
    position: relative;

    padding: 16px 12px;

    @media @mobileBU {
        padding: 20px 28px;
    }
    @media @desktop {
        padding: 20px 32px;
    }

    &__container {
        display: flex;
    }

    &__title {
        margin-bottom: 8px;

        @media @desktopMU {
            margin-bottom: 4px;
        }
    }

    &__title_close-padding {
        padding-right: 20px;
    }

    &__text {
        .commonFont();
        max-width: 100%;
        margin: 0;
    }

    &__text_close-padding {
        padding-right: 20px;
    }

    &__content {
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        align-self: center;
        width: 100%;
        max-width: 212px;
        margin-left: 12px;

        @media @mobileBU {
            max-width: 540px;
        }
        @media @desktop {
            max-width: 640px;
        }
        @media @desktopMU {
            max-width: 840px;
        }
    }

    &__icon-container {
        display: flex;
        flex-shrink: 0;
        align-items: center;
        align-self: flex-start;
        justify-content: center;
        width: 40px;
        min-width: 40px;
        height: 40px;
        border-radius: 50%;
    }

    &__attention-icon,
    &__icon-container svg {
        width: 32px;
        min-width: 32px;
        height: 32px;
    }

    &__link {
        .h5();
        display: inline-flex;
        align-items: center;
        margin-top: 8px;

        cursor: pointer;
    }

    &__right-arrow {
        width: 20px;
        min-width: 20px;
        height: 20px;
        margin-top: 4px;

        fill: @systemBlue;
    }

    &__close {
        position: absolute;
        top: 12px;
        right: 12px;

        display: flex;
        padding: 0;
        border: none;
        border-radius: 50%;

        background: transparent;
        cursor: pointer;
    }

    &__close:hover {
        background-color: @spbSky160;
    }

    &__close-icon {
        width: 20px;
        height: 20px;

        fill: @content;

        @media @mobileBU {
            width: 32px;
            height: 32px;
        }
    }

    &_type_error {
        @{b} {
            &__icon-container {
                background-color: @error;

                svg { /* stylelint-disable-line max-nesting-depth */
                    fill: @fury;
                }
            }
        }
    }

    &_type_success {
        @{b} {
            &__icon-container {
                background-color: @green20;

                svg { /* stylelint-disable-line max-nesting-depth */
                    fill: @green;
                }
            }
        }
    }

    &_type_warning {
        @{b} {
            &__icon-container {
                background-color: @Pantone137C20;

                svg { /* stylelint-disable-line max-nesting-depth */
                    fill: @Pantone137C;
                }
            }
        }
    }

    &_type_info {
        @{b} {
            &__icon-container {
                background-color: @Pantone311C20;

                svg { /* stylelint-disable-line max-nesting-depth */
                    fill: @Pantone311C;
                }
            }
        }
    }

    &_colored {
        @{b} {
            &__icon-container {
                background-color: @base;
            }

            &__close:hover {
                background-color: @base30;
            }
        }
    }

    &_colored&_type_error {
        background-color: @fury80;

        @{b} {
            &__title,
            &__text,
            &__link {
                color: @base;
            }

            &__right-arrow {
                fill: @base;
            }

            &__icon-container {
                background-color: @STCWhite20;

                svg { /* stylelint-disable-line max-nesting-depth */
                    fill: @base;
                }
            }
        }
    }

    &_colored&_type_success {
        background-color: @green20;
    }

    &_colored&_type_warning {
        background-color: @Pantone137C20;
    }

    &_colored&_type_info {
        background-color: @Pantone311C20;
    }
}
