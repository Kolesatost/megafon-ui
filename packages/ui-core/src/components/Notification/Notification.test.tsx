import * as React from 'react';
import Attention from '@megafon/ui-icons/system-16-attention_16.svg';
import { mount, shallow } from 'enzyme';
import Notification, { INotificationProps, NotificationTypes, ShadowTypes } from './Notification';

const props: INotificationProps = {
    className: 'notification',
    classes: {
        root: 'rootClass',
        container: 'containerClass',
        content: 'contentClass',
    },
    type: NotificationTypes.ERROR,
    shadowLevel: ShadowTypes.HOVER,
    link: 'Ссылка',
    href: 'href',
    hasCloseButton: false,
    rel: 'noopener',
    target: '_blank',
    icon: <Attention />,
    title: 'title',
    isColored: true,
    buttonText: 'Кнопка',
};

describe('<Notification />', () => {
    it('renders component', () => {
        const wrapper = shallow(<Notification>Some test text</Notification>);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders colored Notification with Props', () => {
        const wrapper = shallow(<Notification {...props}>Some test text</Notification>);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification with close collapse', () => {
        const wrapper = shallow(<Notification shortText="Короткий текст">Длинный текст</Notification>);
        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification with custom close collapse title', () => {
        const wrapper = shallow(
            <Notification shortText="Короткий текст" closeCollapseTitle="Показать весь текст">
                Длинный текст
            </Notification>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification with open collapse', () => {
        const wrapper = shallow(
            <Notification shortText="Короткий текст" isCollapseOpened>
                Длинный текст
            </Notification>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification with custom open collapse title', () => {
        const wrapper = shallow(
            <Notification isCollapseOpened shortText="Короткий текст" openCollapseTitle="Скрыть часть текста">
                Длинный текст
            </Notification>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification without link, when component has collapse', () => {
        const wrapper = shallow(
            <Notification {...props} shortText="Короткий текст">
                Длинный текст
            </Notification>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('it renders Notification with close collpse after change isCollapseOpened', () => {
        const wrapper = mount(
            <Notification {...props} shortText="Короткий текст" isCollapseOpened>
                Длинный текст
            </Notification>,
        );

        wrapper.setProps({ isCollapseOpened: false });
        wrapper.update();

        expect(wrapper).toMatchSnapshot();
    });

    it('it calls onLinkClick handler', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<Notification {...props} link="link" onLinkClick={onClick} />);
        const link = wrapper.find('.mfui-notification__link');
        link.simulate('click');
        expect(onClick).toBeCalled();
    });

    it('it calls onClick handler', () => {
        const onClose = jest.fn();
        const wrapper = shallow(<Notification {...props} hasCloseButton onClose={onClose} />);
        const closeButton = wrapper.find('.mfui-notification__close');
        closeButton.simulate('click');
        expect(onClose).toBeCalled();
    });

    it('it calls onButtonClick handler', () => {
        const onButtonClick = jest.fn();
        const wrapper = shallow(<Notification {...props} onButtonClick={onButtonClick} />);
        const button = wrapper.find('.mfui-notification__button');
        button.simulate('click');
        expect(onButtonClick).toBeCalled();
    });

    it('it calls onCollapseButtonClick handler', () => {
        const onCollapseButtonClick = jest.fn();

        const wrapper = shallow(
            <Notification shortText="Короткий текст" onCollapseButtonClick={onCollapseButtonClick}>
                Длинный текст
            </Notification>,
        );

        const collapseButton = wrapper.find('.mfui-notification__collapse-button');
        collapseButton.simulate('click');
        expect(onCollapseButtonClick).toBeCalled();
    });
});
