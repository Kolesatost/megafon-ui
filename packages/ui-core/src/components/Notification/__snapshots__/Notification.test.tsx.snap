// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> it renders Notification with close collapse 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <p
          className="mfui-notification__text"
        >
          Короткий текст
          <Collapse
            className="mfui-notification__collapse mfui-notification__collapse_hidden"
            classNameContainer="mfui-notification__collapse-inner"
            isOpened={false}
          >
            Длинный текст
          </Collapse>
        </p>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Подробнее
          <div
            className="mfui-notification__collapse-arrow"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification with custom close collapse title 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <p
          className="mfui-notification__text"
        >
          Короткий текст
          <Collapse
            className="mfui-notification__collapse mfui-notification__collapse_hidden"
            classNameContainer="mfui-notification__collapse-inner"
            isOpened={false}
          >
            Длинный текст
          </Collapse>
        </p>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Показать весь текст
          <div
            className="mfui-notification__collapse-arrow"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification with custom open collapse title 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <p
          className="mfui-notification__text"
        >
          <Collapse
            className="mfui-notification__collapse"
            classNameContainer="mfui-notification__collapse-inner"
            isOpened={true}
          >
            Длинный текст
          </Collapse>
        </p>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Скрыть часть текста
          <div
            className="mfui-notification__collapse-arrow mfui-notification__collapse-arrow_close"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders Notification with open collapse 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <p
          className="mfui-notification__text"
        >
          <Collapse
            className="mfui-notification__collapse"
            classNameContainer="mfui-notification__collapse-inner"
            isOpened={true}
          >
            Длинный текст
          </Collapse>
        </p>
      </div>
      <div
        className="mfui-notification__bottom"
      >
        <button
          className="mfui-notification__collapse-button"
          onClick={[Function]}
          type="button"
        >
          Свернуть
          <div
            className="mfui-notification__collapse-arrow mfui-notification__collapse-arrow_close"
          >
            <test-file-stub />
          </div>
        </button>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> it renders colored Notification with Props 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_error mfui-notification_colored notification rootClass"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="hover"
>
  <div
    className="mfui-notification__container containerClass"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content contentClass"
    >
      <div
        className="mfui-notification__text-container"
      >
        <Header
          as="h5"
          className="mfui-notification__title"
          dataAttrs={
            Object {
              "root": undefined,
            }
          }
        >
          title
        </Header>
        <p
          className="mfui-notification__text"
        >
          Some test text
        </p>
      </div>
      <div
        className="mfui-notification__bottom mfui-notification__bottom_has-button"
      >
        <div
          className="mfui-notification__bottom-block"
        >
          <Button
            className="mfui-notification__button"
            dataAttrs={
              Object {
                "root": undefined,
              }
            }
            disabled={false}
            ellipsis={true}
            showLoader={false}
            sizeAll="small"
            sizeMobile="extra-small"
            theme="white"
          >
            Кнопка
          </Button>
        </div>
      </div>
    </div>
  </div>
</Tile>
`;

exports[`<Notification /> renders component 1`] = `
<Tile
  className="mfui-notification mfui-notification_type_info mfui-notification_colored"
  dataAttrs={
    Object {
      "root": undefined,
    }
  }
  radius="rounded"
  shadowLevel="zero"
>
  <div
    className="mfui-notification__container"
  >
    <div
      className="mfui-notification__icon-container"
    >
      <test-file-stub />
    </div>
    <div
      className="mfui-notification__content"
    >
      <div
        className="mfui-notification__text-container"
      >
        <p
          className="mfui-notification__text"
        >
          Some test text
        </p>
      </div>
    </div>
  </div>
</Tile>
`;
